framework/commercetools/lib/normalize.ts

```diff
@@ -0,0 +1,109 @@
+import type {
+  CommercetoolsProducts,
+  CommerceToolsProductPrice,
+  Product,
+  ProductVariant,
+  CommercetoolsProductVariant,
+  ProductPrice,
+  Images,
+  ProductImage,
+} from '../types/product'
+import type {
+  Cart,
+  CommercetoolsCart,
+  CommercetoolsLineItems,
+  LineItem,
+} from '../types/cart'
+
+function setProductPrice(prices: CommerceToolsProductPrice): ProductPrice {
+  return {
+    value: prices.value.centAmount,
+    currencyCode: prices.value.currencyCode,
+    retailPrice: 0,
+    salePrice: 0,
+    listPrice: 0,
+    extendedListPrice: 0,
+    extendedSalePrice: 0,
+  }
+}
+
+function setImage(images: Images[]): ProductImage[] {
+  return images.map((image) => {
+    return { url: image.url }
+  })
+}
+
+function setVariants(
+  variants: CommercetoolsProductVariant[],
+  published: boolean
+): ProductVariant[] {
+  return variants.map((variant) => {
+    return {
+      id: variant.id,
+      options: [],
+      availableForSale: published,
+    }
+  })
+}
+
+export function normalizeProduct(data: CommercetoolsProducts): Product {
+  const price = setProductPrice(data.masterVariant.prices[0])
+  return {
+    id: data.id,
+    name: data.name && data.name['en'] ? data.name['en'] : '',
+    description:
+      data.description && data.description['en'] ? data.description['en'] : '',
+    sku: data.masterVariant.sku,
+    slug: data.slug['en'] ? data.slug['en'] : '',
+    images: setImage(data.masterVariant.images),
+    variants: setVariants(data.variants, data.published),
+    price,
+    options: [],
+  }
+}
+
+function convertTaxMode(data: CommercetoolsCart): boolean {
+  return data && data.taxMode && data.taxMode === 'Disabled'
+    ? false
+    : data && data.taxMode
+    ? true
+    : false
+}
+export function normalizeCart(data: CommercetoolsCart): Cart {
+  return {
+    id: data.id,
+    customerId: data.customerId,
+    email: data.customerEmail,
+    createdAt: data.createdAt,
+    currency: { code: data.totalPrice.currencyCode },
+    taxesIncluded: convertTaxMode(data),
+    lineItems: data.lineItems.map(normalizeLineItem),
+    lineItemsSubtotalPrice: 0,
+    subtotalPrice: 0,
+    totalPrice: data.totalPrice.centAmount,
```

Comment:
bear in mind that the total price can be including taxes. First we need to check if `data.taxedPrice` exists and if so, set the totalPrice field to `data.taxedPrice.totalGross`. Set the fallback value `data.totalPrice.centAmount` as fallback.